scripts/data_store.py

```diff
@@ -0,0 +1,59 @@
+import os
+import shelve
+import memory
+import message_history
+
+class DataStore():
+
+    def persist_message_history(self, id):
+      return True
+
+    def load_message_history(self, id):
+        return True
+
+    def persist_memory(self ,id):
+        return True
+
+    def load_memory(self, id):
+        return True
+
+instance = DataStore()
+
+class ShelfDataStore(DataStore):
+
+    def __init__(self, path):
+        self.path = path
+        # Ensure path ends with a slash
+        if self.path[-1] != "/":
+            self.path += "/"
+        self.message_history = []
+        self.memory = []
+
+    def persist_message_history(self, id):
+        os.makedirs(f"{self.path}{id}", exist_ok=True)
+        with shelve.open(f"{self.path}{id}/store") as f:
+            f["message_history"] = message_history.message_history
+        return True
+
+    def load_message_history(self, id):
+        try:
+            with shelve.open(f"{self.path}{id}/store") as f:
+                global message_history
+                message_history.set_history(f["message_history"])
+            return True
+        except Exception as e:
+            return e
```

Comment:
This function returns either `True` or `Error`. Both a consistent boolean or an actual exception is better if  you ask me.